docs/docs/how-to/implement_recognizer.md

```diff
@@ -1 +1,305 @@
+`ConstructorCheck` you created in the previous section.
+
+## Referencing the current Entity
+
+Lets take a look at how a check can reference the current entity which is being processed. This is
+something we need in our Singleton recognizer, where we want to check if the class has a field in
+which the instance of the class is stored.
+
+```csharp
+IEnumerable< ICheck > IRecognizer.Create()
+{
+    yield return Class(
+        ...
+        Field(
+            Priority.Knockout,
+            Type(
+                Priority.Knockout,
+                ICheck.GetCurrentEntity
+            )
+        )
+    );
+}
+```
+
+The type matched by a `TypeCheck` depends on the check in which it is nested. In case of a
+`FieldCheck`, it matches the type of the field. Using `ICheck.GetCurrentEntity` we can match the
+current entity which we are trying to match. How this current entity is determined is an
+implementation detail of PatternPal, but regardless of how deeply nested you use
+`ICheck.GetCurrentEntity`, it will always refer to the entity which you are currently trying to
+match.
+
+## Referencing the results of another Check
+
+Now for a more complicated check, lets look at how to use the result of an earlier check. In a
+singleton implementation, you need to have a way to access the instance of the Singleton class. This
+is usually done using a `GetInstance` method. Aside from being `static`, this method should also use
+the field in which the instance is stored. We wrote a check for this field in the previous section,
+but we cannot reference this check currently. To access this check, we can store it inside a
+variable. Checks are only executed when PatternPal encounters them, so storing a check in a
+variable, or defining it in a separate method is perfectly fine, and may lead to more readable
+recognizers. So as a first step, store the `FieldCheck` inside a variable and use it inside the
+`ClassCheck`.
+
+```csharp
+IEnumerable< ICheck > IRecognizer.Create()
+{
+    FieldCheck instanceFieldCheck = Field(
+        Priority.Knockout,
+        Type(
+            Priority.Knockout,
+            ICheck.GetCurrentEntity
+        )
+    );
+
+    yield return Class(
+        Priority.Knockout,
+        Constructor(
+            Priority.Knockout,
+            Modifiers(
+                Priority.Knockout,
+                Modifier.Private
+            )
+        ),
+        instanceFieldCheck
+    );
+}
+```
+
+Now that we have a way to reference the `FieldCheck`, we can also use its results. In this case, we
+want to check that the `GetInstance` method uses the field matched by the `FieldCheck`. To check
+this, we can use a `RelationCheck`. `RelationCheck`s are checks which you can use to check for
+relations between different entities and nodes. Examples of these relations are `Uses`,
+`Implements`, and so on. You can read about all the available relations
+[here](~/docs/technical/syntax_graph.md). Combining the `static` modifier check with this
```

Comment:
I think it is better to reference my section about the checks here, as I explain RelationCheck there, mention all possible relations, and give an example of usage.